### PR TITLE
Let database define id for CollectionJob (Model Transactions Version)

### PR DESCRIPTION
**Pull request by alflennik**
Updates an existing PR (https://github.com/w3c/aria-at-app/pull/925) from @stalgiag to use the new model transactions. A new PR was needed because of the number of conflicts - virtually every line would be considered a conflict (!)

Original PR description:

> This pull request modifies the 'CollectionJob' model so that it has an integer ID that automatically increments as new entries are added. A previous iteration of the Response Collection System specification expected there to be a layer between the app and the job runner that would provide a unique ID tied directly to the job as the identifier for the collection job. The MVP of the automation feature in the app was initially built with this design in mind but part of the way into development this part of the specification changed and as a temporary workaround code was added to the app that manually managed unique auto-incrementing IDs. This PR offloads that responsibility to the DB and simplifies some existing logic. Additionally, previous services that expected CollectionJobs to be created with a defined ID (such as findOrCreateCollectionJob) have been removed.
> 
> Since the new migration in this PR modifies the type of the primary key in the CollectionJob table, I would like to prioritize thorough over speedy review on this one. I tested with a relatively recent dump of production (1/22/24) by running a bot test plan run of the 'Link Example 1 (span element with text content)' using the full system and the simulator.

Since the previous PR was approved I just assigned Stalgia to review it. Stalgia, if you think it's good, I would say go ahead and merge it and then close the other PR.

### CODE REVIEW

**Comment by alflennik** [comment]
Since Stalgia is not on the project this week I assigned Mike!

**Comment by jugglinmike** [comment]
@alflennik I'm seeing now that this pull request targets your open pull request, gh-940. I'm afraid I can't review this without also reviewing the other, and if I did that, I'd be concerned about disrupting the review that's ongoing there. Would you mind requesting my review once that has been approved?

**Comment by alflennik** [comment]
@jugglinmike Sure!

**Comment by alflennik** [comment]
@jugglinmike this is ready for review!

File: server/tests/integration/automation-scheduler.test.js [hunk 1]
```diff
@@ -308,41 +303,42 @@ describe('Automation controller', () => {
...
+                await restartCollectionJobByMutation(job.id, { transaction });
             expect(collectionJob).not.toBe(undefined);
             expect(collectionJob).toEqual({
-                id: jobId,
+                id: job.id,
                 status: 'QUEUED'
             });
             const { collectionJob: storedCollectionJob } =
-                await getTestCollectionJob({ transaction });
-            expect(storedCollectionJob.id).toEqual(jobId);
```
Comment (jugglinmike) [inline server/tests/integration/automation-scheduler.test.js:328]: [The original version of this patch kept this assertion](https://github.com/w3c/aria-at-app/pull/925/files#diff-cb4b6ffb00ae228b144226b4ac4ca796f064c66fa0cae876769c1419aa54034aR294); is this removal intentional?

File: server/tests/integration/automation-scheduler.test.js [hunk 2]
```diff
@@ -429,16 +428,15 @@ describe('Automation controller', () => {
...
+            expect(body.id).toEqual(parseInt(job.id));
             expect(body.status).toEqual('CANCELLED');
             expect(body).toHaveProperty('testPlanRunId');
             expect(body.testPlanRun.testPlanReportId).toEqual(
                 parseInt(testPlanReportId)
             );
 
             const { collectionJob: storedCollectionJob } =
-                await getTestCollectionJob({ transaction });
-            expect(storedCollectionJob.id).toEqual(jobId);
```
Comment (jugglinmike) [inline server/tests/integration/automation-scheduler.test.js:441]: Same question here ([original](https://github.com/w3c/aria-at-app/pull/925/files#diff-cb4b6ffb00ae228b144226b4ac4ca796f064c66fa0cae876769c1419aa54034aR407))

**Review by jugglinmike — COMMENTED** [review]
Just a couple questions about potential coverage regression

**Comment by alflennik** [comment]
@jugglinmike thank you for thoroughly comparing the two versions of this PR and finding those two discrepancies. They've been fixed!

**Review by jugglinmike — APPROVED** [review]
Sure thing
